src/idlenesswatcher.cpp

```diff
@@ -102,8 +104,18 @@ void IdlenessWatcher::setup()
     }
 }
 
-void IdlenessWatcher::timeoutReached(int identifier)
+void IdlenessWatcher::timeoutReached(int identifier,int /*msec*/)
 {
+    // check if disable Idleness when fullscreen is enabled
+    if (mPSettings.isDisableIdlenessWhenFullscreenEnabled()) {
+        WId w = KWindowSystem::activeWindow();
+        KWindowInfo info(w, NET::WMState);
+        if (info.hasState(NET::FullScreen)) {
+            qDebug() << "idleness timeout reached with fullscreen window. Postponing...";
+            return;
```

Comment:
How about a situation when the fullscreen application ends w/o user interaction? Then we will never trigger the idle action. Shouldn't we at least reset the idleness here (`KIdleTime::simulateUserActivity`) to get another `timeoutReached` after a while?

Reply:
Your suggestion is probably the safest path. I'm going to update my patch following it. Thanks :)